tests/system/datastore/test_v3io.py

```diff
@@ -51,51 +59,74 @@ def setup_class(cls):
         }
         with open(cls.test_file_path) as f:
             cls.test_string = f.read()
-        cls.test_dir_path = "/bigdata/v3io_tests"
-        cls.v3io_test_dir_url = "v3io://" + cls.test_dir_path
+        cls.test_dir = "/bigdata/v3io_tests"
+        cls.test_dir_url = "v3io://" + cls.test_dir
+        cls.run_dir = f"{cls.test_dir}/run_{uuid.uuid4()}"
+        cls.profile_name = "v3io_ds_profile"
+        cls.token = os.environ.get("V3IO_ACCESS_KEY")
+        cls.profile = DatastoreProfileV3io(
+            name=cls.profile_name, v3io_access_key=cls.token
+        )
 
     @classmethod
     def teardown_class(cls):
-        dir_data_item = mlrun.get_dataitem(cls.v3io_test_dir_url)
+        dir_data_item = mlrun.get_dataitem(cls.test_dir_url)
         try:
             dir_data_item.delete(recursive=True)
         except Exception:
             pass
         super().teardown_class()
 
-    def setup_method(self, method):
-        self.object_dir_url = f"{self.v3io_test_dir_url}/directory-{uuid.uuid4()}"
-        super().setup_method(method)
+    @pytest.fixture(autouse=True)
+    def setup_before_each_test(self, use_datastore_profile):
+        prefix_path = (
+            f"ds://{self.profile_name}" if use_datastore_profile else "v3io://"
+        )
+        mlrun.datastore.store_manager.reset_secrets()
+        self.run_dir_url = f"{prefix_path}{self.run_dir}"
+        object_file = f"/file_{uuid.uuid4()}.txt"
+        self._object_url = self.run_dir_url + object_file
+        register_temporary_client_datastore_profile(self.profile)
+        if use_datastore_profile:
+            os.environ["V3IO_ACCESS_KEY"] = "wrong_token"
+        else:
+            os.environ["V3IO_ACCESS_KEY"] = self.token
 
     @staticmethod
     def _skip_set_environment():
         return True
 
-    def _get_data_item(self, secrets={}, file_extension="txt"):
-        object_url = f"{self.object_dir_url}/file_{uuid.uuid4()}.{file_extension}"
-        return mlrun.run.get_dataitem(object_url, secrets=secrets), object_url
-
-    def _setup_df_dir(self, first_file_path, second_file_path, file_extension):
-        dataitem_url = f"{self.object_dir_url}/df_{uuid.uuid4()}.{file_extension}"
-
-        uploaded_data_item = mlrun.run.get_dataitem(dataitem_url)
-        uploaded_data_item.upload(first_file_path)
+    def _setup_df_dir(self, file_format, reader):
+        dataframes_dir = f"/{file_format}_{uuid.uuid4()}"
+        dataframes_url = f"{self.run_dir_url}{dataframes_dir}"
+        df1_path = join(self.assets_path, f"test_data.{file_format}")
+        df2_path = join(self.assets_path, f"additional_data.{file_format}")
 
-        dataitem_url = f"{self.object_dir_url}/df_{uuid.uuid4()}.{file_extension}"
-
-        uploaded_data_item = mlrun.run.get_dataitem(dataitem_url)
-        uploaded_data_item.upload(second_file_path)
+        #  upload
```

Comment:
Really not sure what this double spacing is about... Is this the work of some IDE?
```suggestion
        # upload
```